### ISSUE TITLE
[Bug]: Submodule functionality occasionally deletes contents of submodule

### ISSUE THREAD

**Issue by philskents** [Vinzent03/obsidian-git#170]
### Describe the bug

When using submodules obsidian-git occasionally commits the directory as empty, causing all of the contents to be deleted. Seems to be random but I will try to catch some logs.

### Steps to reproduce

- Configure submodule
- Wait

### Operating system

macOS

**Comment by Vinzent03** [comment]
How can files get deleted if I commit a directory as empty? I don't really understand what you think is happening.

**Comment by philskents** [comment]
My apologies @Vinzent03, I did not explain that well, let me try again. So I have a private git repository that houses my notes. In addition I have a submodule in a public repo holding some notes on a specific topic. Every now and again, when obsidian-git backs up the parent repository, I get a commit on the submodule that looks like [this](https://github.com/philskents/allotments-garden/commit/1cebb830eb9b9f3fb519ccdfd282cee30230ce0b). As you can see all of the files are deleted in this commit, even though I have not deleted them myself.

Does that help explain what it is I am seeing?

**Comment by philskents** [comment]
Closed - Ignored
